src/main/webapp/app/file-upload-submission/file-upload-result/file-upload-result.component.ts

```diff
@@ -0,0 +1,24 @@
+import { Component, Input } from '@angular/core';
+import { Feedback } from 'app/entities/feedback';
+import { Result } from 'app/entities/result';
+import { partition } from 'lodash';
+
+@Component({
+    selector: 'jhi-file-upload-result',
+    templateUrl: './file-upload-result.component.html',
+})
+export class FileUploadResultComponent {
+    public feedbacks: Feedback[];
+    public generalFeedback: Feedback;
+
+    @Input()
+    public set result(result: Result) {
+        if (!result) {
+            return;
+        }
+        const groupedFeedback = partition(result.feedbacks, feedback => feedback.credits > 0);
```

Comment:
Very nice!

You could improve it further:
`const [feedbackWithCredits, feedbackWithoutCredits] = partition(result.feedbacks, feedback => feedback.credits > 0);`

This naming helps to understand what you are partitioning.

Reply:
Good suggestion! Indeed this is more clear